src/lib/y2network/autoinst_profile/interface_section.rb

```diff
@@ -0,0 +1,320 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/autoinst_profile/section_with_attributes"
+
+module Y2Network
+  module AutoinstProfile
+    # This class represents an AutoYaST <interface> section under <interfaces>
+    #
+    #  <interface>
+    #    <bootproto>static</bootproto>
+    #    <broadcast>127.255.255.255</broadcast>
+    #    <device>lo</device>
+    #    <firewall>no</firewall>
+    #    <ipaddr>127.0.0.1</ipaddr>
+    #    <netmask>255.0.0.0</netmask>
+    #    <network>127.0.0.0</network>
+    #    <prefixlen>8</prefixlen>
+    #    <startmode>nfsroot</startmode>
+    #    <usercontrol>no</usercontrol>
+    #  </interface>
+    #
+    # @see InterfacesSection
+    class InterfaceSection < SectionWithAttributes
+      def self.attributes
+        [
+          { name: :bootproto },
+          { name: :broadcast },
+          { name: :device },
+          { name: :name }, # has precedence over device
+          { name: :ipaddr },
+          { name: :remote_ipaddr },
+          { name: :netmask },
+          { name: :network }, # TODO: what it is? looks like ipaddr with applied prefix
+          { name: :prefixlen }, # has precedence over netmask
+          { name: :startmode },
+          { name: :ifplugd_priority },
+          { name: :usercontrol }, # no longer used, ignored
+          { name: :dhclient_set_hostname },
+          { name: :bonding_master },
+          { name: :bonding_slave0 },
+          { name: :bonding_slave1 },
+          { name: :bonding_slave2 },
+          { name: :bonding_slave3 },
+          { name: :bonding_slave4 },
+          { name: :bonding_slave5 },
+          { name: :bonding_slave6 },
+          { name: :bonding_slave7 },
+          { name: :bonding_slave8 },
+          { name: :bonding_slave9 },
```

Comment:
isn't it nice in comparison to boring ```bridge_ports``` ? :-/

Reply:
both are horrible. It should be array. But I cannot change it due to backward compatibility.

Reply:
I know, I was just crying.